x/erc20/keeper/genesis.go

```diff
@@ -0,0 +1,121 @@
+package keeper
+
+import (
+	"fmt"
+	"math/big"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/evmos/evmos/v16/x/erc20/types"
+	"github.com/evmos/evmos/v16/x/evm/statedb"
+)
+
+// SetGenesisTokenPairs stores in state the provided token pairs in genesis
+func (k Keeper) SetGenesisTokenPairs(ctx sdk.Context, pairs []types.TokenPair) error {
+	for i, pair := range pairs {
+		contractAddr := pair.GetERC20Contract()
+		coinMeta, err := k.getGenesisTokenPairMeta(ctx, pair, i)
+		if err != nil {
+			return fmt.Errorf("error while generating metadata for genesis pair denom %s. %w", pair.Denom, err)
+		}
+		stateDB := statedb.New(ctx, k.evmKeeper, statedb.TxConfig{})
+		code, err := k.generateContractCode(ctx, stateDB, coinMeta, contractAddr)
+		if err != nil {
+			return fmt.Errorf("error while getting contract code for genesis pair denom %s. %w", pair.Denom, err)
+		}
+		stateDB.SetCode(contractAddr, code)
+
+		if err := stateDB.Commit(); err != nil {
```

Comment:
it's not clear why do we need to call commit. Please document the logic. Ideally we should only use EVM keeper funcs

Reply:
The problem of doing that is that we cannot control the address of the contract. We have to create an ERC20 contract in the provided address at genesis

If we use here an EVM keeper to create the contract, the contract address will depend on the nonce of the erc20 module address

Reply:
ok, makes sense to add a comment in the code to make it more maintainable